examples/tree_2d_dgsem/elixir_hypdiff_harmonic_nonperiodic.jl

```diff
@@ -10,21 +10,22 @@ equations = HyperbolicDiffusionEquations2D()
 @inline function initial_condition_harmonic_nonperiodic(x, t,
                                                         equations::HyperbolicDiffusionEquations2D)
     # elliptic equation: -ν Δϕ = 0 in Ω, u = g on ∂Ω
-    if t == 0.0
-        phi = 1.0
-        q1 = 1.0
-        q2 = 1.0
+    RealT = eltype(x)
+    if t == 0
+        phi = one(RealT)
+        q1 = one(RealT)
+        q2 = one(RealT)
     else
-        C = inv(sinh(pi))
-        sinpi_x1, cospi_x1 = sincos(pi * x[1])
-        sinpi_x2, cospi_x2 = sincos(pi * x[2])
-        sinh_pix1 = sinh(pi * x[1])
-        cosh_pix1 = cosh(pi * x[1])
-        sinh_pix2 = sinh(pi * x[2])
-        cosh_pix2 = cosh(pi * x[2])
+        C = inv(sinh(convert(RealT, pi)))
+        sinpi_x1, cospi_x1 = sincos(convert(RealT, pi) * x[1])
+        sinpi_x2, cospi_x2 = sincos(convert(RealT, pi) * x[2])
+        sinh_pix1 = sinh(convert(RealT, pi) * x[1])
+        cosh_pix1 = cosh(convert(RealT, pi) * x[1])
+        sinh_pix2 = sinh(convert(RealT, pi) * x[2])
+        cosh_pix2 = cosh(convert(RealT, pi) * x[2])
```

Comment:
How about these - do they have corresponding `pi` functions?

Reply:
No (at least not in Base).

Reply:
Thanks